src/FSharp.Core/quotations.fsi

```diff
@@ -1439,6 +1439,16 @@ module Patterns =
     [<CompiledName("ForIntegerRangeLoopPattern")>]
     val (|ForIntegerRangeLoop|_|): input: Expr -> (Var * Expr * Expr * Expr) option
 
+    /// <summary>An active pattern to recognize expressions that represent loops over integer ranges with non-one step</summary>
+    ///
+    /// <param name="input">The input expression to match against.</param>
+    ///
+    /// <returns>When successful, the pattern binds the value, start, step, finish and body of the input expression</returns>
+    ///
+    /// <example-tbd></example-tbd>
+    [<CompiledName("ForIntegerRangeLoopWithStepPattern")>]
+    val (|ForIntegerRangeLoopWithStep|_|): input: Expr -> (Var * Expr * Expr * Expr * Expr) option
```

Comment:
Optimizations shouldn't really cause additions to the quotation API.  If we were to add something, we would need to add a corresponding quotation construction method, e.g. similar to https://github.com/dotnet/fsharp/blob/9978a14a48812b18f0c5b3caa33602df8749fe78/src/FSharp.Core/quotations.fs#L2632